erc721-locker/contracts/ERC721Locker.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import { IERC721Metadata } from "@openzeppelin/contracts/token/ERC721/IERC721Metadata.sol";
+import { IERC165 } from "@openzeppelin/contracts/introspection/IERC165.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { Borsh } from "rainbow-bridge/contracts/eth/nearbridge/contracts/Borsh.sol";
+import { AdminControlled } from "rainbow-bridge/contracts/eth/nearbridge/contracts/AdminControlled.sol";
+import { ProofDecoder } from "rainbow-bridge/contracts/eth/nearprover/contracts/ProofDecoder.sol";
+
+import { IERC721Locker } from "./interfaces/IERC721Locker.sol";
+import { INearProver, Locker } from "./Locker.sol";
+
+contract ERC721Locker is IERC721Locker, Locker, AdminControlled {
+    using Strings for uint256;
+
+    /*
+     *     bytes4(keccak256('name()')) == 0x06fdde03
+     *     bytes4(keccak256('symbol()')) == 0x95d89b41
+     *     bytes4(keccak256('tokenURI(uint256)')) == 0xc87b56dd
+     *
+     *     => 0x06fdde03 ^ 0x95d89b41 ^ 0xc87b56dd == 0x5b5e139f
+     */
+    bytes4 private constant _INTERFACE_ID_ERC721_METADATA = 0x5b5e139f;
+
+    uint constant PAUSE_FINALISE_FROM_NEAR = 1 << 0;
+    uint constant PAUSE_TRANSFER_TO_NEAR = 1 << 1;
+
+    event Locked (
+        address indexed token,
+        address indexed sender,
+        string tokenId,
+        string accountId,
+        string tokenURI
+    );
+
+    event Unlocked (
+        address token,
+        uint256 tokenId,
+        address recipient
+    );
+
+    // This reverse lookup is needed as token IDs on Near are strings and therefore this is needed for unlocking
+    // NFT contract address -> string token ID -> uint256
+    mapping(address => mapping(string => uint256)) public stringTokenIdToUnitForNft;
```

Comment:
Do we really need this mapping, given it is just the *identity* function?

We can convert between string and uint instead of storing this info

Reply:
I removed this mapping and added a function to convert a string to unit [here](https://github.com/Shard-Labs/rainbow-non-fungible-token-connector/blob/dev/ERC721Locker/contracts/ERC721Locker.sol#L177)